libimage/push.go

```diff
@@ -50,7 +49,44 @@ func (r *Runtime) Push(ctx context.Context, source, destination string, options
 		destination = resolvedSource
 	}
 
-	logrus.Debugf("Pushing image %s to %s", source, destination)
+	// If specified to push --all-tags, look them up and iterate.
+	if options.AllTags {
+
+		// Do not allow : for tags, other than specifying transport
+		d := strings.TrimPrefix(destination, "docker://")
+		if strings.ContainsAny(d, ":") {
+			return nil, fmt.Errorf("tag can't be used with --all-tags/-a")
+		}
+
+		namedRepoTags, err := image.NamedTaggedRepoTags()
```

Comment:
Wait. What is actually the semantics of the `AllTags` option? Don’t say “like docker push”, _actually write it out_ (and document for the `PushOptions` field), please.

AFAICS Docker’s implementation finds all local images that match the provided repository (even if there are multiple different images), and pushes all of them.

This one finds a _single_ image, looks up the tags within that provided repository that point at that single image, and pushes only those.

That’s a completely different behavior. Which one is this intended to implement?

Reply:
The former: push all images for the matched repository.

I'll take the examples provided to shore up the logic and integrate the comments within the existing Push runtime.